### ISSUE TITLE
False negatives for consider-iterating-dictionary

### ISSUE THREAD

**Issue by rogalski** [PyCQA/pylint#1247]
### Steps to reproduce
Run pylint from master branch on following code:
```
DICT = {'a': 1, 'b': 2}

# correct emit
COMP = [k * 2 for k in DICT.keys()]

# incorrect not-emit
COMP1 = [k * 2 for k in DICT.keys()] + [k * 3 for k in DICT.keys()]
COMP2, COMP3 = [k * 2 for k in DICT.keys()], [k * 3 for k in DICT.keys()]
SOME_TUPLE = ([k * 2 for k in DICT.keys()], [k * 3 for k in DICT.keys()])
```

### Current behavior
Only one message (in line 4) is emitted.

### Expected behavior
There should be messages emitted in lines 7, 8 and 9.

### pylint --version output
```
(py36) mac-mini:pylint rogalski$ python -m pylint --version
__main__.py 2.0.0, 
astroid 1.5.0
Python 3.6.0 (default, Dec 26 2016, 14:24:38) 
[GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.42.1)]
```

This one also affects my patch #1245. Recently merged #1244 solved issue only partially.
